pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunctionTest.java

```diff
@@ -0,0 +1,295 @@
+    FileUtils.deleteQuietly(new File(getIndexDirPath(ENABLE_NULL_SEGMENT_NAME)));
+    for (int i = 0; i < NUM_ROWS; i++) {
+      _intSVValues[i] = RANDOM.nextInt();
+    }
+    List<GenericRow> rows = new ArrayList<>(NUM_ROWS);
+    for (int i = 0; i < NUM_ROWS; i++) {
+      Map<String, Object> map = new HashMap<>();
+      map.put(INT_SV_COLUMN, _intSVValues[i]);
+      if (isEqualRow(i)) {
+        map.put(INT_SV_NULL_COLUMN, _intSVValues[i]);
+      } else if (isNotEqualRow(i)) {
+        map.put(INT_SV_NULL_COLUMN, _intSVValues[i] + 1);
+      } else if (isNullRow(i)) {
+        map.put(INT_SV_NULL_COLUMN, null);
+      }
+      GenericRow row = new GenericRow();
+      row.init(map);
+      rows.add(row);
+    }
+
+    Schema schema = new Schema.SchemaBuilder().addSingleValueDimension(INT_SV_COLUMN, FieldSpec.DataType.INT)
+        .addSingleValueDimension(INT_SV_NULL_COLUMN, FieldSpec.DataType.INT).build();
+    _enableNullDataSourceMap = getDataSourceMap(schema, rows, ENABLE_NULL_SEGMENT_NAME);
+    _enableNullProjectionBlock = getProjectionBlock(_enableNullDataSourceMap);
+    _disableNullDataSourceMap = getDataSourceMap(schema, rows, DISABLE_NULL_SEGMENT_NAME);
+    _disableNullProjectionBlock = getProjectionBlock(_disableNullDataSourceMap);
+  }
+
+  protected void testTransformFunction(ExpressionContext expression, boolean[] expectedValues,
+      ProjectionBlock projectionBlock)
+      throws Exception {
+    int[] intValues = getTransformFunctionInstance(expression).transformToIntValuesSV(projectionBlock);
+    long[] longValues = getTransformFunctionInstance(expression).transformToLongValuesSV(projectionBlock);
+    float[] floatValues = getTransformFunctionInstance(expression).transformToFloatValuesSV(projectionBlock);
+    double[] doubleValues = getTransformFunctionInstance(expression).transformToDoubleValuesSV(projectionBlock);
+    String[] stringValues = getTransformFunctionInstance(expression).transformToStringValuesSV(projectionBlock);
+    for (int i = 0; i < NUM_ROWS; i++) {
+      Assert.assertEquals(intValues[i] == 1, expectedValues[i]);
+      Assert.assertEquals(longValues[i] == 1, expectedValues[i]);
+      Assert.assertEquals(floatValues[i] == 1, expectedValues[i]);
+      Assert.assertEquals(doubleValues[i] == 1, expectedValues[i]);
+      Assert.assertEquals(stringValues[i], Boolean.toString(expectedValues[i]));
+    }
+  }
+
+  private TransformFunction getTransformFunctionInstance(ExpressionContext expression) {
+    return TransformFunctionFactory.get(expression, _enableNullDataSourceMap);
+  }
+
+  // Test that left column of the operator has null values and right column is not null.
+  @Test
+  public void testDistinctFromLeftNull()
+      throws Exception {
+    ExpressionContext isDistinctFromExpression = RequestContextUtils.getExpression(
+        String.format("%s(%s, %s)", IS_DISTINCT_FROM, INT_SV_NULL_COLUMN, INT_SV_COLUMN));
```

Comment:
the expression should be
`String.format("%s IS DISTINCT FROM %s", NT_SV_NULL_COLUMN, INT_SV_COLUMN)` 
since the correct SQL syntax is without the underscore.

Reply:
Done.